tiledb/sm/c_api/tiledb.cc

```diff
@@ -2778,16 +2780,29 @@ int32_t tiledb_query_condition_combine(
     return TILEDB_OOM;
   }
 
-  if (SAVE_ERROR_CATCH(
-          ctx,
-          left_cond->query_condition_->combine(
-              *right_cond->query_condition_,
-              static_cast<tiledb::sm::QueryConditionCombinationOp>(
-                  combination_op),
-              (*combined_cond)->query_condition_))) {
-    delete (*combined_cond)->query_condition_;
-    delete *combined_cond;
-    return TILEDB_ERR;
+  if (combination_op == TILEDB_NOT) {
```

Comment:
It might be useful here to have a comment that we are partially evaluating the tree and applying NOT via DeMorgan's theorem.

Reply:
Will do.